_clients/dot-net-conventions.md

```diff
@@ -0,0 +1,90 @@
+
+- **CloudConnectionPool**
+
+`CloudConnectionPool` is a subclass of `SingleNodeConnectionPool` that takes a Cloud ID and credentials. Like `SingleNodeConnectionPool`, `CloudConnectionPool` does not support sniffing or pinging.
+
+- **StaticConnectionPool**
+
+`StaticConnectionPool` is used for a small cluster if you don't want to turn on sniffing to learn about cluster topology. `StaticConnectionPool` does not support sniffing, but can support pinging.
+
+- **SniffingConnectionPool**
+
+`SniffingConnectionPool` is a subclass of `StaticConnectionPool`. It is thread safe and supports sniffing and pinging. `SniffingConnectionPool` can be reseeded at run time, and you can specify node roles when seeding.
+
+- **StickyConnectionPool**
+
+`StickyConnectionPool` is set up to return the first live node which then persists between requests. It can be seeded using an enumerable of `Uri` or `Node` objects. `StickyConnectionPool` does not support sniffing, but supports pinging.
+
+- **StickySniffingConnectionPool**
+
+`StickySniffingConnectionPool` is a subclass of `SniffingConnectionPool`. Like `StickyConnectionPool`, it returns the first live node which then persists between requests. `StickySniffingConnectionPool` supports sniffing and sorting so that each instance of your application can favor a different node. Nodes have weights associated with them, and can be sorted by weight.
+
+## Retries
+
+If a request does not succeed, it is automatically retried. By default, the number of retries is the number of nodes in your cluster known to OpenSearch.Client. The number of retries is also limited by the timeout parameter, so OpenSearch.Client retries requests as many times as possible within the timeout period. 
+
+To set the maximum number of retries, specify this number in the `MaximumRetries` property on the `ConnectionSettings` object.
+
+```cs
+var settings = new ConnectionSettings(connectionPool).MaximumRetries(5);
+```
+
+You can also set a `RequestTimeout` that specifies a timeout for one single request, and a `MaxRetryTimeout` that specifies the time limit for all retry attempts. In the example below, `RequestTimeout` is set to 4 seconds, and `MaxRetryTimeout` is set to 12 seconds, so the maximum number of attempts for a query is 3. 
+
+```cs
+var settings = new ConnectionSettings(connectionPool)
+            .RequestTimeout(TimeSpan.FromSeconds(4))
+            .MaxRetryTimeout(TimeSpan.FromSeconds(12));
+```
+
+## Failover
+
+If you are using a connection pool with multiple nodes, a request is retried if it returns a 502 (Bad Gateway), 503 (Service Unavailable) or 504 (Gateway Timeout) HTTP error response code. If the response code is an error code in the 400–501 or 505–599 range, the request is not retried.
```

Comment:
Insert the serial comma after "(Service Unavailable)". If we are referring to two ranges, "range" should be "ranges".